scripts/redistributeValoraRewards/proposeSafeAddRewardsTx.ts

```diff
@@ -0,0 +1,157 @@
+ * @param rewardPoolAddress The RewardPool contract address (Valora redistribution reward pool)
+ * @param rewardAmounts The rewards distributed for the given campaign
+ * @param valoraRewards The amount of valora rewards to add (as string or bigint)
+ * @param excludeReferrerIds The referrer ids to exclude from the rewards
+ * @param rewardsFilename The filename of the rewards file (used as idempotency key)
+ * @param networkId The NetworkId for the Safe Transaction Service
+ * @param alchemyKey The Alchemy API key
+ * @param dryRun Whether to dry run the transaction
+ */
+export async function proposeSafeAddRewardsTx({
+  safeAddress: rawSafeAddress,
+  rewardPoolAddress: rawRewardPoolAddress,
+  rewardAmounts,
+  valoraRewards,
+  excludeReferrerIds,
+  rewardsFilename,
+  networkId,
+  alchemyKey,
+  dryRun,
+}: {
+  safeAddress: Address
+  rewardPoolAddress: string
+  rewardAmounts: Array<{
+    referrerId: Address
+    rewardAmount: string
+  }>
+  valoraRewards: bigint
+  excludeReferrerIds: Address[]
+  rewardsFilename: string
+  networkId: NetworkId
+  alchemyKey: string
+  dryRun: boolean
+}) {
+  const safeConfig = NETWORK_ID_TO_SAFE_CONFIG[networkId]
+  const alchemyRpcUrl = NETWORK_ID_TO_ALCHEMY_RPC_URL[networkId]
+  const safeAddress = getAddress(rawSafeAddress)
+  const rewardPoolAddress = getAddress(rawRewardPoolAddress)
+
+  if (!safeConfig) {
+    throw new Error(`No Safe config found for networkId: ${networkId}`)
+  }
+
+  if (!alchemyRpcUrl) {
+    throw new Error(`No Alchemy RPC URL found for networkId: ${networkId}`)
+  }
+
+  const protocolKit = await Safe.init({
+    provider: `${alchemyRpcUrl}${alchemyKey}`,
+    safeAddress,
+  })
+
+  const nonValoraReferrersWithRewards = rewardAmounts.filter(
+    (reward) =>
+      !excludeReferrerIds.includes(reward.referrerId) &&
+      BigInt(reward.rewardAmount) > 0,
+  )
+
+  if (nonValoraReferrersWithRewards.length === 0) {
+    throw new Error('No non-valora referrers with rewards')
+  }
+
+  const rewardAmount =
+    valoraRewards / BigInt(nonValoraReferrersWithRewards.length)
+
+  // 1. Encode calldata for claimReward or deposit
+  const data = encodeFunctionData({
+    abi: rewardPoolAbi,
+    functionName: 'addRewards',
+    args: [
+      nonValoraReferrersWithRewards.map((referrer) => ({
+        user: referrer.referrerId,
+        amount: rewardAmount,
+        // use the rewards filename as the idempotency key to avoid adding double rewards
+        idempotencyKey: keccak256(
+          toBytes(`${rewardsFilename}-${referrer.referrerId}`),
+        ),
+      })),
+      [],
+    ],
+  })
+
+  const safeTransactionData = {
+    to: rewardPoolAddress,
+    value: '0',
+    data,
+    operation: OperationType.Call,
+  }
+  // 2. Prepare Safe transaction data
+  const safeTx = await protocolKit.createTransaction({
+    transactions: [safeTransactionData],
+  })
+
+  const safeTxHash = await protocolKit.getTransactionHash(safeTx)
+
+  console.info(
```

Comment:
This log format was meant for a CF. This could be really noisy in a script. We should keep this and other logs minimal